listener/src/event_handler/flux_transition_handler.rs

```diff
@@ -0,0 +1,209 @@
+use super::*;
+
+use dfusion_core::models::{PendingFlux, State};
+use dfusion_core::models::util::{pop_u8_from_log_data, pop_u256_from_log_data, pop_h256_from_log_data, to_value};
+
+use graph::components::store::{EntityFilter, Store};
+use graph::data::store::Entity;
+use std::fmt;
+use web3::types::U256;
+
+#[derive(Clone)]
+pub struct FluxTransitionHandler {
+    store: Arc<Store>,
+}
+
+impl FluxTransitionHandler {
+    pub fn new(store: Arc<Store>) -> Self {
+        FluxTransitionHandler{
+            store
+        }
+    }
+}
+
+impl Debug for FluxTransitionHandler {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "FluxTransitionHandler")
+    }
+}
+
+enum FluxTransitionType {
+    Deposit = 0,
+    Withdraw = 1,
+}
+
+impl From<u8> for FluxTransitionType {
+    fn from(tranition_type: u8) -> Self {
+        match tranition_type {
+            0 => FluxTransitionType::Deposit,
+            1 => FluxTransitionType::Withdraw,
+            _ => panic!("Unknown transition type: {}", tranition_type),
+        }
+    }
+}
+
+impl EventHandler for FluxTransitionHandler {
+    fn process_event(
+        &self,
+        logger: Logger,
+        _block: Arc<EthereumBlock>,
+        _transaction: Arc<Transaction>,
+        log: Arc<Log>
+    ) -> Result<Vec<EntityOperation>, Error> {
+        let mut data = log.data.0.clone();
+        let transition_type: FluxTransitionType = pop_u8_from_log_data(&mut data).into();
+        let state_index = pop_u256_from_log_data(&mut data).saturating_sub(U256::one());
+        let new_state_hash = pop_h256_from_log_data(&mut data);
+        let slot = pop_u256_from_log_data(&mut data);
+
+        info!(logger, "Received Flux State Transition Event");
+
+        let account_query = util::entity_query(
+            "AccountState", EntityFilter::Equal("stateIndex".to_string(), to_value(&state_index))
+        );
+        let mut account_state = State::from(self.store
+            .find_one(account_query)?
+            .ok_or_else(|| failure::err_msg(format!("No state record found for index {}", &state_index)))?
+        );
+
+        match transition_type {
+            FluxTransitionType::Deposit => {
+                let deposit_query = util::entity_query(
+                    "Deposit", EntityFilter::Equal("slot".to_string(), to_value(&slot))
+                );
+                let deposits = self.store
+                    .find(deposit_query)?
+                    .into_iter()
+                    .map(PendingFlux::from)
+                    .collect::<Vec<PendingFlux>>();
+                account_state.apply_deposits(&deposits);
```

Comment:
sexy!

Reply:
However, I can't seem to find `apply_deposits`... I assume it is implemented for state and was done in a different PR.

Reply:
Yes, it was https://github.com/gnosis/dex-services/pull/192